documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -1024,4 +1029,180 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
+
+**Why the rule is important**: This will provide a consistent experience to the user.
+
+**How to fix the violation**: Ensure that when a tracked resource has list by resource group and subscription operations, the x-ms-pageable extension values are same for both operations.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2017" />R2017 PutRequestResponseScheme
+**Output Message**: A PUT operation request body schema should be the same as its 200 response schema, to allow reusing the same entity between GET and PUT. If the schema of the PUT request body is a superset of the GET response body, make sure you have a PATCH operation to make the resource updatable. Operation: '{0}' Request Model: '{1}' Response Model: '{2}'
+
+**Description**: The request & reponse('200') schema of the 'put' operation must be same.
+
+**Why the rule is important**: This will provide a consistent experience to the user.
+
+**How to fix the violation**: Ensure the request & reponse('200') schema of the 'put' operation must be same.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2062" />R2062 XmsResourceInPutResponse
+**Output Message**: The 200 response model for an ARM PUT operation must have x-ms-azure-resource extension set to true in its hierarchy. Operation: '{0}' Model: '{1}'.
+
+**Description**: The 200 response model for an ARM PUT operation must have x-ms-azure-resource extension set to true in its hierarchy. Operation: '{0}' Model: '{1}'.
+
+**Why the rule is important**: This will ensure that the 'put' operation actually returns a resource.
+
+**How to fix the violation**: Ensure that the 200 response model for an ARM PUT operation must have x-ms-azure-resource extension set to true in its hierarchy.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2016" />R2016 PatchBodyParametersSchema
+**Output Message**: Properties of a PATCH request body must not be {0}. PATCH operation: '{1}' Model Definition: '{2}' Property: '{3}'
+
+**Description**: A request parameter of the Patch Operation must not have a required/default value.
+
+**Why the rule is important**: This will ensure that the existing properties are not accidentally overwritten.
```

Comment:
I'm not sure this is the reason for this rule to exist. I believe that PATCH is to replace the value of certain properties and nothing should be required, since it would mean that property would always need to be overwritten, the case of default, is because if not passed, calling the PATCH operation I believe would reset its value to the default value. Please confirm what I've stated before and/or find out the reason why this is needed.